osu.Desktop/OsuGameDesktop.cs

```diff
@@ -54,19 +56,29 @@ public override void SetHost(GameHost host)
         private void dragDrop(DragEventArgs e)
         {
             // this method will only be executed if e.Effect in dragEnter gets set to something other that None.
-            var dropData = e.Data.GetData(DataFormats.FileDrop) as object[];
+            var dropData = (object[])e.Data.GetData(DataFormats.FileDrop);
             var filePaths = dropData.Select(f => f.ToString()).ToArray();
-            ImportBeatmapsAsync(filePaths);
+
+            if (filePaths.All(f => Path.GetExtension(f) == @".osz"))
+                Task.Run(() => BeatmapDatabase.Import(filePaths));

```

Comment:
Why not just `.Where` and change the if statements to `.Any`? Then you can drag a mix of files and not have it do nothing at all.